src/main/java/com/pnu/dev/radioserviceapi/util/mapper/VideoMapper.java

```diff
@@ -0,0 +1,62 @@
+package com.pnu.dev.radioserviceapi.util.mapper;
+
+import com.pnu.dev.radioserviceapi.client.dto.search.ItemYoutubeSearchResponse;
+import com.pnu.dev.radioserviceapi.client.dto.videos.ItemYoutubeVideosResponse;
+import com.pnu.dev.radioserviceapi.dto.response.VideoDto;
+import com.pnu.dev.radioserviceapi.mongo.LiveBroadcastContent;
+import com.pnu.dev.radioserviceapi.mongo.Video;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class VideoMapper {
+
+    public Video itemVideosResponseToVideo(ItemYoutubeVideosResponse itemVideosResponse) {
+
+        return Video.builder()
+                .id(itemVideosResponse.getId())
+                .title(itemVideosResponse.getSnippet().getTitle())
+                .description(itemVideosResponse.getSnippet().getDescription())
+                .build();
+    }
+
+    public List<VideoDto> itemSearchResponseListToVideoDtoList(List<ItemYoutubeSearchResponse> youtubeSearchResponse) {
+
+        return youtubeSearchResponse.stream().map(this::itemSearchResponseToVideoDto).collect(Collectors.toList());
+    }
+
+    private VideoDto itemSearchResponseToVideoDto(ItemYoutubeSearchResponse itemSearchResponse) {
+
+        return VideoDto.builder()
+                .id(itemSearchResponse.getId().getVideoId())
+                .title(itemSearchResponse.getSnippet().getTitle())
+                .description(itemSearchResponse.getSnippet().getDescription())
+                .publishedAt(itemSearchResponse.getSnippet().getPublishedAt())
+                .liveBroadcastContent(LiveBroadcastContent
+                        .valueOf(itemSearchResponse.getSnippet().getLiveBroadcastContent().toUpperCase()))
+                .build();
+    }
+
+    public Page<VideoDto> videoPageToVideoDtoPage(Page<Video> videoPage) {
+
+        return new PageImpl<>(videoListToVideoDtoList(videoPage.getContent()));
```

Comment:
Would this page, that you return, still hold appropriate current page number and total pages number?
Use simple dto as a return type with data and `currentPageNumber` + `totalPagesNumber` fields.

Reply:
Yes, Page class contains such fields as pageNumber, totalResults, totalPageNumber, isFirst, isLast etc

Reply:
I mean, will the object, that you return from this method contain the same values of this fields as the object you accept as a parameter?